_rules/aria-role-permitted-j7zzqr.md

```diff
@@ -0,0 +1,167 @@
+---
+id: j7zzqr
+name: ARIA role is permitted for the element.
+rule_type: atomic
+description: |
+  This rule checks that WAI-ARIA roles are allowed for the element they are specified on.
+accessibility_requirements:
+  html-aria:docconformance:
+    title: ARIA in HTML, 4. Document conformance requirements for use of ARIA attributes in HTML
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: satisfied
+input_aspects:
+  - Accessibility Tree
+  - DOM Tree
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+---
+
+## Applicability
+
+This rule applies to any [HTML element][namespaced element] which has an [explicit semantic role][explicit role].
+
+## Expectation
+
+For each test target, its [explicit semantic role][explicit role] is allowed on this element, according to the [ARIA in HTML specification][aria in html document conformance].
+
+## Assumptions
+
+There are no assumptions.
+
+## Accessibility Support
+
+There are no accessibility support issues known.
+
+## Background
+
+[ARIA in HTML][aria in html document conformance] also defines the [implicit semantic role][implicit role] of each element. Setting the [explicit role][] as the same as the [implicit one][implicit role] is not recommended but nonetheless allowed. This rule doesn't use that in any of its test cases.
+
+This rules apply to every element, even if they are not [included in the accessibility tree][]. This is because the roles of `none` or `presentation` are only allowed on certain elements. If the rule was only looking at content [included in the accessibility tree][], it wouldn't flag incorrect use of these roles. 
```

Comment:
i mean, yes - if the element is hidden form the accessibility tree then sure, no one can access it / there can't be an issue.

the counter point being that if it's rendered in the DOM but marked as hidden, is that because at some point it would be expected that a user could then unhide that content to interact with it?  Could excluding something just because it's temporarily hidden not cause a bunch of gotchas for developers who might run a scan, it comes up clean, and then they still get bugs because someone had the audacity to try and use their UI? :)

but seriously, i can see the rational to just expose this as a recommendation / needs review - in the hidden state, as long as if it becomes shown it then would be marked as a violation.  it just seems like a delay in flagging some issues because maybe someone did a hack-job at 'removing' an errant element, and _that_ shouldn't be called out?